File: instructor-client/src/app/model/entity/dto/server-metrics-dto.ts
export interface ServerMetricsDto {
  cpu_usage_percent: number,
  total_disk_space_in_bytes: number,
  remaining_disk_space_in_bytes: number,
  saved_screenshots_size_in_bytes: number,
  saved_videos_size_in_bytes: number,
  max_available_memory_in_bytes: number,
  total_used_memory_in_bytes: number,
}

File: server/src/main/java/at/htl/franklyn/server/boundary/Dto/ServerMetricsDto.java
package at.htl.franklyn.server.boundary.Dto;

public record ServerMetricsDto(
        double cpuUsagePercent,
        double totalDiskSpaceInBytes,
        double remainingDiskSpaceInBytes,
        double savedScreenshotsSizeInBytes,
        double maxAvailableMemoryInBytes,
        double totalUsedMemoryInBytes,
        long alphaUploadsCount,
        double alphaUploadAvgDurationMs,
        double alphaUploadBytesTotal,
        long alphaUploadErrors,
        long betaUploadsCount,
        double betaUploadAvgDurationMs,
        double betaUploadBytesTotal,
        double betaChangedPixelsTotal,
        long betaUploadErrors,
        long requestedAlphaFrames,
        long screenshotFetchCount,
        double screenshotFetchAvgMs,
        long screenshotScaledFetchCount,
        double screenshotScaledFetchAvgMs,
        long screenshotErrors,
        long videoDownloadsCount,
        double videoDownloadAvgMs,
        long videoDownloadErrors,
        long screenshotsFileCount
) {
}
